powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/CallbackUrlBehavior.java

```diff
@@ -458,110 +459,60 @@ private Map<String, Object> prepareCallbackDataOperation(CallbackUrlEntity callb
-            builder.responseTimeout(configuration.getHttpResponseTimeout());
-        }
-        if (configuration.getHttpMaxIdleTime() != null) {
-            builder.maxIdleTime(configuration.getHttpMaxIdleTime());
-        }
-        if (configuration.getHttpProxyEnabled()) {
-            final DefaultRestClient.ProxyBuilder proxyBuilder = builder.proxy().host(configuration.getHttpProxyHost()).port(configuration.getHttpProxyPort());
-            if (configuration.getHttpProxyUsername() != null) {
-                proxyBuilder.username(configuration.getHttpProxyUsername()).password(configuration.getHttpProxyPassword());
-            }
-        }
-        final CallbackUrlAuthenticationEntity authentication = callbackUrlEntity.getAuthentication();
-        final CallbackUrlAuthenticationEntity.Certificate certificateAuth = authentication.getCertificate();
-        if (certificateAuth != null && certificateAuth.isEnabled()) {
-            final DefaultRestClient.CertificateAuthBuilder certificateAuthBuilder = builder.certificateAuth();
-            if (certificateAuth.isUseCustomKeyStore()) {
-                certificateAuthBuilder.enableCustomKeyStore()
-                        .keyStoreLocation(certificateAuth.getKeyStoreLocation())
-                        .keyStorePassword(certificateAuth.getKeyStorePassword())
-                        .keyAlias(certificateAuth.getKeyAlias())
-                        .keyPassword(certificateAuth.getKeyPassword());
-            }
-            if (certificateAuth.isUseCustomTrustStore()) {
-                certificateAuthBuilder.enableCustomTruststore()
-                        .trustStoreLocation(certificateAuth.getTrustStoreLocation())
-                        .trustStorePassword(certificateAuth.getTrustStorePassword());
-            }
-        }
-        final CallbackUrlAuthenticationEntity.HttpBasic httpBasicAuth = authentication.getHttpBasic();
-        if (httpBasicAuth != null && httpBasicAuth.isEnabled()) {
-            builder.httpBasicAuth()
-                    .username(httpBasicAuth.getUsername())
-                    .password(httpBasicAuth.getPassword());
-        }
-        return builder.build();
+    private CallbackUrlEventEntity createAndSaveCallbackUrlEventEntity(final CallbackUrlEntity callbackUrlEntity, final Map<String, Object> callbackData) {
+        final LocalDateTime timestampNow = LocalDateTime.now();
+
+        final CallbackUrlEventEntity callbackUrlEventEntity = new CallbackUrlEventEntity();
+        callbackUrlEventEntity.setCallbackUrlEntity(callbackUrlEntity);
+        callbackUrlEventEntity.setCallbackData(callbackData);
+        callbackUrlEventEntity.setIdempotencyKey(UUID.randomUUID().toString());
+        callbackUrlEventEntity.setTimestampCreated(timestampNow);
+        callbackUrlEventEntity.setTimestampLastCall(timestampNow);
+        callbackUrlEventEntity.setAttempts(0);
+        callbackUrlEventEntity.setStatus(CallbackUrlEventStatus.PROCESSING);
```

Comment:
After some thought.
To protect the situation when callback receiver doesn't have idempotency - for whatever reasons) We should set the status as INSTANT (and change the respective part in the postCallback and cleaning jobs. 

In this particular case ,we do not need to store the event at all (but it might be handy as log supplement) but this might bring too big complexity

Reply:
Instead of introducing another state and increasing complexity, the rerun logic is now based on a new column `timestamp_rerun_after` which is set to `now + rerun period` when an event gets into PROCESSING state (i.e. when first created or when taken from DB as PEDNING).
Events, which are configured with max attempts = 1, does not get `timestamp_rerun_after` and therefore are not part of the reset job.